test_regress/t/t_vpi_module.cpp

```diff
@@ -48,24 +48,55 @@ unsigned int main_time = 0;
         return __LINE__; \
     }
 
+#define CHECK_RESULT_Z(got) \
+    if (got) { \
+        printf("%%Error: %s:%d: GOT = !NULL  EXP = NULL\n", FILENM, __LINE__); \
+        return __LINE__; \
+    }
+
 #define CHECK_RESULT_CSTR(got, exp) \
     if (strcmp((got), (exp))) { \
         printf("%%Error: %s:%d: GOT = '%s'   EXP = '%s'\n", FILENM, __LINE__, \
                (got) ? (got) : "<null>", (exp) ? (exp) : "<null>"); \
         return __LINE__; \
     }
 
+void modDump(vpiHandle it, int n) {
+    while (vpiHandle hndl = vpi_scan(it)) {
```

Comment:
We need to free up handles after use, e.g. use TestVpiHandle here and other places get a handle.

Please run with --sanitize to check they are all ok (after merging in master).  Also beware #2706.  Feel free to push even without #2706 cleaned up and I'll fix this test later.